docs/developer/testing/TESTING.md

```diff
@@ -1,61 +1,123 @@
+| `OADP_CRED_FILE` | The path to credentials file for backupLocations | `/var/run/oadp-credentials/new-aws-credentials` | true |
+| `OADP_BUCKET` | The bucket name to store backups | `OADP_BUCKET_FILE` file content | true |
+| `CI_CRED_FILE` | The path to credentials file for snapshotLocations | `/Users/drajds/.aws/.awscred` | true |
+| `VSL_REGION` | The region of snapshotLocations | - | true |
+| `BSL_REGION` | The region of backupLocations | `us-east-1` | false |
+| `OADP_TEST_NAMESPACE` | The namespace where OADP operator is installed | `openshift-adp` | false |
+
+The expected format for `OADP_CRED_FILE` and `CI_CRED_FILE` files is:
 ```
 [<INSERT_PROFILE_NAME>]
 aws_access_key_id=<access_key>
 aws_secret_access_key=<secret_key>
 ```
+> **Note:** If you use one profile name different from `default` for backupLocations, also change `BSL_AWS_PROFILE` environment variable. snapshotLocations must use `default` profile name.
+
+To set the environment variables, run
+```sh
+export OADP_CRED_FILE=<path_to_backupLocations_credentials_file>
+export OADP_BUCKET=<bucket_name>
+export CI_CRED_FILE=<path_to_snapshotLocations_credentials_file>
+export VSL_REGION=<snapshotLocations_region>
+# non required
+export BSL_REGION=<backupLocations_region>
+export OADP_TEST_NAMESPACE=<test_namespace>
+```
 
-`OADP_S3_BUCKET`:
-```json
-{
-  "velero-bucket-name": <bucket>
-}
+It is also possible to set the environment variables during make command call. Example
+```sh
+OADP_CRED_FILE=<path_to_backupLocations_credentials_file> \
+OADP_BUCKET=<bucket_name> \
+CI_CRED_FILE=<path_to_snapshotLocations_credentials_file> \
+VSL_REGION=<snapshotLocations_region> \
+BSL_REGION=<backupLocations_region> \
+OADP_TEST_NAMESPACE=<test_namespace> \
+make test-e2e
 ```
 
-## Run all e2e tests
+### Azure setup
+
+TODO
+
+### GCP setup
+
+TODO
+
+## Run tests
+
+To run all E2E tests for your provider, run
 ```bash
-$ make test-e2e
+make test-e2e
 ```
-## Run selected test
-You can run a particular e2e test(s) by placing an `F` at the beginning of a
-`Describe`, `Context`, and `It`.
+### Run selected test
 
+You can run a particular e2e test(s) by placing an `F` at the beginning of Ginkgo objects. Example
+
+```go
+FDescribe("test description", func() { ... })
+FContext("test scenario", func() { ... })
+FIt("the assertion", func() { ... })
+...
 ```
- FDescribe("test description", func() { ... })
- FContext("test scenario", func() { ... })
- FIt("the assertion", func() { ... })
+
+These need to be removed to run all specs. Checks [Ginkgo docs](https://onsi.github.io/ginkgo/) for more info.
+
+## Clean up
+
+To clean environment after running E2E tests, run
+```bash
+oc delete backup -n $OADP_TEST_NAMESPACE --all
+oc delete backuprepository -n $OADP_TEST_NAMESPACE --all
+oc delete downloadrequest -n $OADP_TEST_NAMESPACE --all
+oc delete podvolumerestore -n $OADP_TEST_NAMESPACE --all
+oc delete restore -n $OADP_TEST_NAMESPACE --all # This fails because of finalizer :(
```

Comment:
Do you know how to remove finalizer from the CLI?

Reply:
https://stackoverflow.com/a/52820472/4590470